dali/imgcodec/parsers/webp.cc

```diff
@@ -12,19 +13,144 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include "dali/imgcodec/parsers/webp.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+
+struct RiffHeader {
+  std::array<uint8_t, 4> riff_text;
+  std::array<uint8_t, 4> file_size;
+  std::array<uint8_t, 4> webp_text;
+};
+static_assert(sizeof(RiffHeader) == 12);
+
+using vp8_header_t = std::array<uint8_t, 4>;
+
+// Simple file format (lossy)
+// https://datatracker.ietf.org/doc/html/rfc6386#section-9.1
+struct WebpLossyHeader {
+  std::array<uint8_t, 4> chunk_size;
+  std::array<uint8_t, 3> frame_tag;
+  std::array<uint8_t, 3> sync_code;
+};
+static_assert(sizeof(WebpLossyHeader) == 10);
+
+// Simple file format (lossless)
+// https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#2_riff_header
+struct WebpLosslessHeader {
+  std::array<uint8_t, 4> chunk_size;
+  uint8_t signature_byte;
+};
+static_assert(sizeof(WebpLosslessHeader) == 5);
```

Comment:
Nice idea with the typed structures. Do you think we can (or maybe should) add a comment, that shows the layout of the expected data in the docstring somewhere? Specifically to signify that those structures are packed one after another.

Something like:
```
Header layout:

RiffHeader|vp8_header_t|<WebpLossyHeader/WebpLosslessHeader>

The WebpLossyHeader and WebpLosslessHeader are selected based on the "VP8 " and "VP8L" in vp8_header_t.
"VP8X" in vp8_header_t signifies WebP extended file format which is not supported.
```

Reply:
Seems like a good idea